api/v1/controllers/collectors.js

```diff
@@ -77,6 +79,59 @@ function reEncryptSGContextValues(sg, authToken, timestamp) {
 }
 
 /**
+ * Find the matching generator template and attach it to the generator
+ *
+ * @param  {Object} sg - Sample generator object
+ * @return {Object} Sample generator with generatorTemplate attribute set to the
+ *  full matching generator template object.
+ */
+function attachTemplate(sg) {
+  const { name, version } = sg.generatorTemplate;
+  return GeneratorTemplate.getSemverMatch(name, version)
+  .then((gt) => {
+    if (sg.context && gt.contextDefinition) {
+      sg = JSON.parse(JSON.stringify(sg));
+      sg.generatorTemplate = gt;
+      return sg;
+    }
+  });
+}
+
+/**
+ * TODO: delete this once the auto-registration has been set up and the heartbeat
+ * has been updated to use it
+ */
+/**
+ * Register a collector. Access restricted to Refocus Collector only.
+ *
+ * @param {IncomingMessage} req - The request object
+ * @param {ServerResponse} res - The response object
+ * @param {Function} next - The next middleware function in the stack
+ */
+function postCollector(req, res, next) {
+  const collectorToPost = req.swagger.params.queryBody.value;
+  const resultObj = { reqStartTime: req.timestamp };
+  const toPost = req.swagger.params.queryBody.value;
+  helper.model.create(toPost)
+  .then((o) => {
+    if (helper.loggingEnabled) {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime;
+      utils.logAPI(req, resultObj, o);
```

Comment:
"utils" is not defined.

Reply:
I only added this as a temporary solution to make the heartbeat tests work, and they do, so that if statement must never be true. I think it's fine as-is.